cli/server/server_test.go

```diff
@@ -0,0 +1,228 @@
+	require.Equal(t, netmode.TestNet, cfg.ProtocolConfiguration.Magic)
+}
+
+func TestHandleLoggingParams(t *testing.T) {
+	testLog := "./test.log"
+	defer func() {
+		os.Remove(testLog)
+	}()
+
+	t.Run("default", func(t *testing.T) {
+		set := flag.NewFlagSet("flagSet", flag.ExitOnError)
+		ctx := cli.NewContext(cli.NewApp(), set, nil)
+		cfg := config.ApplicationConfiguration{
+			LogPath: testLog,
+		}
+		logger, err := handleLoggingParams(ctx, cfg)
+		require.NoError(t, err)
+		require.True(t, logger.Core().Enabled(zap.InfoLevel))
+		require.False(t, logger.Core().Enabled(zap.DebugLevel))
+	})
+
+	t.Run("debug", func(t *testing.T) {
+		set := flag.NewFlagSet("flagSet", flag.ExitOnError)
+		set.Bool("debug", true, "")
+		ctx := cli.NewContext(cli.NewApp(), set, nil)
+		cfg := config.ApplicationConfiguration{
+			LogPath: testLog,
+		}
+		logger, err := handleLoggingParams(ctx, cfg)
+		require.NoError(t, err)
+		require.True(t, logger.Core().Enabled(zap.InfoLevel))
+		require.True(t, logger.Core().Enabled(zap.DebugLevel))
+	})
+}
+
+func TestInitBCWithMetrics(t *testing.T) {
+	defer func() {
+		os.RemoveAll("./chains/")
+	}()
+
+	set := flag.NewFlagSet("flagSet", flag.ExitOnError)
+	set.String("config-path", "../../config", "")
+	set.Bool("testnet", true, "")
+	set.Bool("debug", true, "")
+	ctx := cli.NewContext(cli.NewApp(), set, nil)
+	cfg, err := getConfigFromContext(ctx)
+	require.NoError(t, err)
+	logger, err := handleLoggingParams(ctx, cfg.ApplicationConfiguration)
+	chain, prometheus, pprof, err := initBCWithMetrics(cfg, logger)
+	require.NoError(t, err)
+	defer chain.Close()
+	defer prometheus.ShutDown()
+	defer pprof.ShutDown()
+	require.Equal(t, netmode.TestNet, chain.GetConfig().Magic)
+}
+
+func TestDumpDB(t *testing.T) {
+	testDump := "./test_dump.acc"
+
+	t.Run("too low chain", func(t *testing.T) {
+		defer func() {
+			os.RemoveAll("./chains/")
+			os.RemoveAll(testDump)
+		}()
+		set := flag.NewFlagSet("flagSet", flag.ExitOnError)
+		set.String("config-path", "../../config", "")
+		set.Bool("privnet", true, "")
+		set.Bool("debug", true, "")
+		set.Int("start", 0, "")
+		set.Int("count", 5, "")
+		set.String("out", testDump, "")
+		ctx := cli.NewContext(cli.NewApp(), set, nil)
+		err := dumpDB(ctx)
+		require.Error(t, err)
+	})
+
+	t.Run("positive", func(t *testing.T) {
+		defer func() {
+			os.RemoveAll("./chains/")
+			os.RemoveAll(testDump)
+		}()
+		set := flag.NewFlagSet("flagSet", flag.ExitOnError)
+		set.String("config-path", "../../config", "")
+		set.Bool("privnet", true, "")
+		set.Bool("debug", true, "")
+		set.Int("start", 0, "")
+		set.Int("count", 1, "")
+		set.String("out", testDump, "")
+		ctx := cli.NewContext(cli.NewApp(), set, nil)
+		err := dumpDB(ctx)
+		require.NoError(t, err)
+	})
+}
+
+func TestRestoreDB(t *testing.T) {
+	testDump := "./test_dump.acc"
+	saveDump := "./dump.acc"
+	defer func() {
+		os.RemoveAll("./chains/")
+		os.RemoveAll(testDump)
+		os.RemoveAll(saveDump)
```

Comment:
`ioutil.TempDir`? Maybe change working directory to that temp dir. But then wipe it and forget it.